hash.ts

```diff
@@ -10,7 +10,148 @@ export type Description = Record<string, string | string[]> & {
   eNS?: Record<string, Record<string, string>>;
 };
 
-const referenceLookups: Record<string, (e: Element) => Element[]> = {};
+/* currently only works if doName is set, GSSE support is missing */
+function findFCDAReferences(e: Element): Element[] {
+  const ldInst = e.getAttribute('ldInst');
+  const prefix = e.getAttribute('prefix');
+  const lnClass = e.getAttribute('lnClass');
+  const lnInst = e.getAttribute('lnInst');
+  const doName = e.getAttribute('doName');
+  const daName = e.getAttribute('daName');
+  const fc = e.getAttribute('fc');
+
+  if (!doName) {
+    return [];
+  }
+
+  const references = [] as Element[];
+
+  const server = e.closest('Server');
+  if (!server) {
+    return [];
+  }
```

Comment:
Interesting, "pure client LNs" don't have a `Server` element.

![image](https://github.com/user-attachments/assets/b8d267cd-24dc-4670-8450-ad015d180c86)

I think this means we need to consider this alternative path. From Ed 2.1:

> If neither a router, nor a clock, nor a server, nor a LN list is specified, the access point may
only be used by client LNs in the same IED to access the bus to which it is connected. This is
typical for a process bus access point of a bay level device, where the LNs offer their data via
a server to the station bus only.

I've not seen actual IEDs implemented like this so if there is some confusion we must check with @JakobVogelsang

Reply:
I've read this part, of course, but have not been able to make heads or tails of it. Now that you're quoting the paragraph alongside the diagram it's starting to make more sense. Does this case only apply if we don't have `ldInst` set at all, or do we always check and ignore `ldInst` in this case?

Reply:
Jacob says this should not be relevant because FCDA can only reference data used in control blocks, so it will always point inside a server.